From 1778c4c466e473c5a66761ecb44bc4292259d26e Mon Sep 17 00:00:00 2001
From: "guoping.li" <guoping.li@amlogic.com>
Date: Wed, 15 Nov 2023 10:22:41 +0800
Subject: [PATCH 2/3] Multimedia: separate build output files for mediactrlsrc
 [1/1]

PD#SWPL-146132

Problem:
build output file *.o/*.so/bin is in the same file as the source code
it cause git error
hdmictrl cannot run

Solution:
change the makefile and bb, set build output file to build/temp/work...

Verify:
local

Signed-off-by: guoping.li <guoping.li@amlogic.com>
Change-Id: I2b4a58401969f15e07b864f61cc5e0e22d7a9b19
---
 hdmictrl/Makefile | 12 ++++++++++--
 makefile_bin.inc  | 41 +++++++++++++++++++++++++++++++++++++++++
 2 files changed, 51 insertions(+), 2 deletions(-)
 create mode 100644 makefile_bin.inc

diff --git a/hdmictrl/Makefile b/hdmictrl/Makefile
index ff3b61b..9e2789e 100644
--- a/hdmictrl/Makefile
+++ b/hdmictrl/Makefile
@@ -1,5 +1,4 @@
 TARGET = hdmictrl
-SRCS = hdmictrl.c
 EXTRA_CFLAGS += ""
 CFLAGS = -I../include
 CFLAGS += $(EXTRA_CFLAGS) -Wno-stringop-truncation
@@ -7,4 +6,13 @@ LDFLAGS += -L../
 LDLIBS += -lm -lrt -lpthread -ltvclient
 
 
-include ../makefile.inc
+SRCS = \
+	hdmictrl.cc
+
+OBJDIR = objs
+OBJS = $(patsubst %.cc,$(OBJDIR)/%.o,$(SRCS))
+
+
+
+include ../makefile_bin.inc
+
diff --git a/makefile_bin.inc b/makefile_bin.inc
new file mode 100644
index 0000000..79186eb
--- /dev/null
+++ b/makefile_bin.inc
@@ -0,0 +1,41 @@
+CFLAGS += -Wall -Werror=jump-misses-init -fPIC -Os
+LDFLAGS += -Wall -Wextra -fPIC -Os
+
+
+CFLAGS += \
+	-I$(STAGING_DIR)/usr/include/
+
+LDFLAGS += -L$(STAGING_DIR)/usr/lib/ \
+			-L$(OUT_DIR) \
+			$(LDLIBS)
+
+
+all: $(TARGET)
+
+$(info "LDLIBS : $(LDLIBS)")
+
+$(info "TARGET_CFLAGS : $(TARGET_CFLAGS)")
+$(info "CFLAGS : $(CFLAGS)")
+$(info "LDFLAGS : $(LDFLAGS)")
+$(info "OUT_DIR : $(OUT_DIR)")
+$(info "STAGING_DIR : $(STAGING_DIR)")
+$(info "TARGET_DIR : $(TARGET_DIR)")
+
+
+$(TARGET): $(OUT_DIR)/$(OBJS)
+	@echo " [LINK] $@"
+	@$(CXX) $(LDFLAGS) $+ $(LDLIBS) -o $(OUT_DIR)/$@
+
+
+$(OUT_DIR)/$(OBJDIR)/%.o: %.c
+	@echo " [Compile] $<"
+	@mkdir -p $(OUT_DIR)/$(OBJDIR)
+	@$(CC) $(CFLAGS) -o $@ -c $<
+
+
+.PHONY:clean
+
+clean:
+	@echo " [CLEAN] ..."
+	@rm -rf $(OUT_DIR)/$(OBJDIR) $(OUT_DIR)/$(TARGET)
+
-- 
2.25.1

